sdk/core/azure-core/azure/core/pipeline/transport/base.py

```diff
@@ -337,20 +337,34 @@ def _format_url_section(template, **kwargs):
                 template = "/".join(components)
         # No URL sections left - returning None
 
+    @staticmethod
+    def _urljoin(base_url, stub_url):
+        # type: (str, str) -> str
+        """Append to end of base URL without losing query parameters.
+
+        :param str base_url: The base URL.
+        :param str stub_url: Section to append to the end of the URL path.
+        :returns: The updated URL.
+        :rtype: str
+        """
+        parsed = urlparse(base_url)
+        parsed = parsed._replace(path=parsed.path + '/' + stub_url)
+        return parsed.geturl()
+
     def format_url(self, url_template, **kwargs):
         # type: (str, Any) -> str
         """Format request URL with the client base URL, unless the
         supplied URL is already absolute.
 
         :param str url_template: The request URL to be formatted if necessary.
         """
-        url = self._format_url_section(url_template, **kwargs)
+        url = PipelineClientBase._format_url_section(url_template, **kwargs)
         if url:
             parsed = urlparse(url)
             if not parsed.scheme or not parsed.netloc:
                 url = url.lstrip('/')
                 base = self._base_url.format(**kwargs).rstrip('/')
-                url = urljoin(base + '/', url)
+                url = PipelineClientBase._urljoin(base, url)
```

Comment:
Would suggest `super` here and above instead of invoking directly on`PipelineClientBase`. Generally speaking, if `super` is used anywhere (as we do for `__init__` calls), then it should be used everywhere, to ensure a consistent MRO. Practically speaking, probably never an issue in this case, but if `super` is used then a reader doesn't have to stop and wonder why it *isn't* used.

Reply:
> Would suggest `super` here and above instead of invoking directly on`PipelineClientBase`. Generally speaking, if `super` is used anywhere (as we do for `__init__` calls), then it should be used everywhere, to ensure a consistent MRO. Practically speaking, probably never an issue in this case, but if `super` is used then a reader doesn't have to stop and wonder why it _isn't_ used.

I see we use self in other places.

Reply:
`self` is even more preferable, if there is not an implementation to avoid on this class. I assumed the intent was to specifically call an implementation further up the inheritance structure.

Reply:
Actually I missed from the diff that these were static methods. So this is fine (or making them standalone helper functions would also be good, I think the draft guidelines suggest to avoid static methods)

Reply:
Thanks @bryevdv, @xiangyan99 
Except this attribute isn't on the super class....
I can replace with `self`, though I generally don't like using `self` on static methods because `self` isn't passed in...

Reply:
Thanks @bryevdv - didn't realize we had a guideline on that - moved them to standalone functions